### PR TITLE
Milestone 4: Implementing logic

### PR DESCRIPTION
**Pull request by 8Bts**
In this branch, I added 2 files.
 - `calculate.js` for button logics and mutating the state of the App component.
 - `operate.js` for performing a mathematical operations using the `big.js` package.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import operate from './operate';
+
+const calculate = (data, btnName) => {
+  let {
+    total, next, operation, content, priorNext, priorOp,
+  } = data;
+
```
Comment (uche-inyama) [inline src/logic/calculate.js:7]: From the project requirement on number 7: the **data prop** should have three properties: total, next, and operation. 
Here's the link: https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376018/

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      break;
+    case '/':
+      result = a.div(b);
+      break;
+    default:
+      break;
+  }
+  return parseFloat(result);
+};
+
```
Comment (uche-inyama) [inline src/logic/operate.js:25]: Operate.js should have a case for **%** according to no 14 of the project requirements. Here is the link:
https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376018/

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi Rashid Mammadli :slightly_smiling_face: 

The requirements for this milestone were almost met. Just few refactoring, to do.
Kindly check the inline comments below.

#### That's it, refactor those lines of code and submit another form for a code review.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Comment by 8Bts** [comment]
Hi @uche-inyama!
Thanks for your code review. I tried my best to meet the requirements mentioned by you. However, I could not fulfill it to 100%. The reason for this is that the `data` parameter cannot contain only 3 properties.
I moved 2 properties to the module's global scope but I still need one additional property to have in the state object of the `App` component. This is the `content` property, which is responsible to store the **calculation path** and it is very important in order to have normal UX/UI.

 _Best,
Rashid_

**Comment by mimipeshy** [comment]
## STATUS : APPROVED 🥳 👯 

Hello team @8Bts 👋🏾 👋🏾 congratulations on making it this far in this milestone 🥇 

#### Highlights

- You were professional in communicating on the areas you have worked on and why ✔️  
- You have worked on all changes requested by the previous TSE ✔️ 

With that, I approve of this project 😄 

Regards
__Peris Ndanu __
[GitHub ](https://github.com/mimipeshy)| [LinkedIn ](https://www.linkedin.com/in/peris-ndanu/)| [Twitter](https://twitter.com/pygirl254)

**Pull request merged**